### DISCUSSION TITLE
seaborn.objects: z order of layers

### DISCUSSION THREAD

**Discussion by stanmart** [mwaskom/seaborn#3233] [Q&A] [answered]
Is there any way to determine the z order of the various layers on a plot with multiple objects? For example, `Dot`s seems to be plotted earlier than `Dash`es, regardless of the order of the `.add` method calls. In certain cases, it would be nice to have control over which layer comes first, e.g. to make sure that elements representing summary statistics are easily visible.

**Comment by mwaskom** [comment, marked as answer]
`seaborn.objects` currently does not do anything different than matplotlib, which assigns different default zorders to different artist classes. I agree that it would make sense to have the zorder appear more responsive to the order that layers are added. For now, you can use the `artist_kws` argument of any `Mark` to assign the artists comprising the mark a specific zorder.

**Comment by stanmart** [comment]
Thank you, passing the `artist_kws` is indeed a great solution for the moment. If you ever decide to extend seaborn's API to handle this without going down to matplotlib, I am happy to (try to) help.

**Comment by matulad** [comment]
Hi @mwaskom ,

change proposal - let's add those two lines to the `so.Plot().add()` method (e.g. to rows 541 or so):

```{python}
zorder = len(self._layers) + 1
mark.artist_kws["zorder"] = mark.artist_kws.get("zorder", zorder)
```

Simple example - let's say I would like to plot points over a line:

```{python}
(
  so.Plot(tips.sample(20), "total_bill", "tip")
  .add(so.Line(linewidth=3))
  .add(so.Dot(pointsize=50, color="red"))
)
```

Without the two lines:

![zorder_wo](https://user-images.githubusercontent.com/107509750/219490179-c37c6b1c-8585-4413-9705-814d8ed278de.png)

With the lines:

![zorder_with](https://user-images.githubusercontent.com/107509750/219490221-e5976edf-6aaf-4749-b093-1c6919ed23f9.png)


Happy to hear your feedback. Thanks.
